### DISCUSSION TITLE
Implementing serverless Postgres

### DISCUSSION THREAD

**Discussion by eqoram** [supabase/supabase#9174] [Feature Requests]
Hi folks,

First of all thanks a lot to the Supabase team for building such an awesome product which really helped me having fun again when programming for the web. :)

I just stumbled across this cool open source project which also might be of interest to you. Neon.tech (https://github.com/neondatabase/neon) is a serverless Postgres database and looks like it offers a lot of advantages for Supabase. Especially the scalability aspect is amazing and could bring in my opinion next level autoscaling to the product. From my understanding it could also help out with having data closer to the user and I think kiwicopple already mentioned in a podcast I recently listened to that edge computing is something they want to dig into even further.

So what is your opinion on this? Do you think an integration of the Neon serverless Postgres database into the Supabase tech stack makes sense? Also I would love to know if the Supabase team is already looking into it as they are probably already aware of it. :)

**Comment by igl** [comment]
I love supabase and neon also sounds amazing. 

There is a [changelog episode about neon](https://changelog.com/podcast/510). Sounds like the patches that Neon requires are going upstream into PG and Supabase can take advantage of those too.

Besides neon being serverless, I am hyped about the database branching.
I have only seen that feature in Datomic and PlanetScale (mySQL) and it would be awesome to have it in Supabase.

**Comment by dalkommatt** [comment]
+1 serverless and branching

**Comment by kylerummens** [comment]
Any update to this? I just heard about Neon today as Vercel just launched [Vercel Postgres](https://vercel.com/storage/postgres) built on Neon.

Would love autoscaling and branching.

**Comment by abhay187** [comment]
Branching will make development a lot easy. Seriously I hate running superbase containers during development.

**Comment by abhay187** [comment]
@awalias please respond.

**Comment by GitIgnoreMaybe** [comment]
And up

**Comment by webclinic017** [comment]
any update on this - supabase + neon - it would be advantageous together

**Comment by eqoram** [comment]
This looks promising: https://github.com/kiwicopple/serverless-postgres
